content/5_Plat/RURQ.mdx

```diff
@@ -153,48 +322,43 @@ situations might take place after any update:
 Given the mechanics behind the `push_down` function, all you need is a regular
 range-sum segment tree to solve the question.
 
+### Implementation
+**Time Complexity:** $\mathcal{O}(N + Q\log{N})$
+
+<LanguageSection>
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
 using ll = long long;
 
-/**
- * Represents the type of lazy update being done.
- * NONE = if there's no lazy update to be performed.
- */
 enum QueryType { ADD, SET, NONE };
 
 struct Query {
 	QueryType type = NONE;
 	ll val = 0;
 };
 
-/**
- * T = tree node type, which will be long long
- * We use the Query struct to manage updates.
- */
-template <class T> class LazySegtree {
+class LazySegtree {
   private:
 	const int sz;
-	vector<T> tree;      // tree[i] = sum of this node's range
+	vector<ll> tree;     // tree[i] = sum of this node's range
```

Comment:
why'd you remove the generic

Reply:
i think it's sorta unnecessary, especially because this impl is less of a template

Reply:
i mean in the purs module we always used the templates, and i think it's nice to stay consistent

Reply:
k sure